src/gui/csvImport/CsvImportWidget.cpp

```diff
@@ -0,0 +1,294 @@
+    connect(m_ui->buttonBox, SIGNAL(rejected()), this, SLOT(reject()));
+}
+
+void CsvImportWidget::comboChanged(int comboId) {
+    QComboBox* currentSender = qobject_cast<QComboBox*>(m_comboMapper->mapping(comboId));
+    if (currentSender->currentIndex() != -1)
+        //this line is the one that actually updates GUI table
+        m_parserModel->mapColumns(currentSender->currentIndex(), comboId);
+    updateTableview();
+}
+
+void CsvImportWidget::skippedChanged(int rows) {
+    m_parserModel->setSkippedRows(rows);
+    updateTableview();
+}
+
+CsvImportWidget::~CsvImportWidget() {}
+
+void CsvImportWidget::configParser() {
+    m_parserModel->setBackslashSyntax(m_ui->checkBoxBackslash->isChecked());
+    m_parserModel->setComment(m_ui->comboBoxComment->currentText().at(0));
+    m_parserModel->setTextQualifier(m_ui->comboBoxTextQualifier->currentText().at(0));
+    m_parserModel->setCodec(m_ui->comboBoxCodec->currentText());
+    m_parserModel->setFieldSeparator(m_ui->comboBoxFieldSeparator->currentText().at(0));
+}
+
+void CsvImportWidget::updateTableview() {
+    m_ui->tableViewFields->resizeRowsToContents();
+    m_ui->tableViewFields->resizeColumnsToContents();
+
+    for (int c = 0; c < m_ui->tableViewFields->horizontalHeader()->count(); ++c)
+        m_ui->tableViewFields->horizontalHeader()->setSectionResizeMode(c, QHeaderView::Stretch);
+}
+
+void CsvImportWidget::updatePreview() {
+
+    m_ui->labelSizeRowsCols->setText(m_parserModel->getFileInfo());
+    m_ui->spinBoxSkip->setValue(0);
+    m_ui->spinBoxSkip->setMaximum(qMax(0, m_parserModel->rowCount() - 1));
+
+    int i;
+    QStringList list(tr("Not present in CSV file"));
+
+    for (i = 1; i < m_parserModel->getCsvCols(); i++) {
+        QString s = QString(tr("Column ")) + QString::number(i);
+        list << s;
+    }
+    m_comboModel->setStringList(list);
+
+    i=1;
+    Q_FOREACH (QComboBox* b, m_combos) {
+        if (i < m_parserModel->getCsvCols())
+            b->setCurrentIndex(i);
+        else
+            b->setCurrentIndex(0);
+        ++i;
+    }
+}
+
+void CsvImportWidget::load(const QString& filename, Database* const db) {
+    //QApplication::processEvents();
+    m_db = db;
+    m_parserModel->setFilename(filename);
+    m_ui->labelFilename->setText(filename);
+    Group* group = m_db->rootGroup();
+    group->setUuid(Uuid::random());
+    group->setNotes(tr("Imported from CSV file").append("\n").append(tr("Original data: ")) + filename);
+    parse();
+}
+
+void CsvImportWidget::parse() {
+    configParser();
+    QApplication::setOverrideCursor(Qt::WaitCursor);
+    //QApplication::processEvents();
+    bool good = m_parserModel->parse();
+    updatePreview();
+    QApplication::restoreOverrideCursor();
+    if (good)
+        //four newline are provided to avoid resizing at every Positive/Warning switch
+        m_ui->messageWidget->showMessage(QString("\n\n").append(tr("CSV syntax seems in good shape !"))
```

Comment:
messageWidget height changes with the text, and all the rest of the window content will follow, sliding up and down. This happens every single time you change any of the Encoding parameters, so every format which is not "standard" (i.e. every format which doesn't come from a password manager) will probably require some tweaking triggering this behaviour.

Use the widget with naked messages - no newline padding - and within a couple of swirls your eyes will implore harakiri. If you can play around with this db you will understand what I mean (note the second row with a field equal to `"2\""`:

"Account","Login Name","Password","Web Site","Comments"
"test1","2\\"","pwd","URL","some notes..."
"test","€èéç","hahaha","",""
work/group1;acc3;log3;pass3;ws3;com3;field3;lco3
work/group2/subgroup3;acc4;log4;pass4;ws4;com4;field4;lco4
fun/basket;acc5;log5;pass5;ws5;com5;field5;lco5
fun/basket;acc6;log5;pass5;ws5;com5;field5;lco5
fun/basket;acc7;log5;pass5;ws5;com5;field5;lco5
abbaz\as;accZ;log5;pass5;ws5;com5;field5;lco5

Do you think a large, reassuring green pad is so bad? Or that is it really worst than an elevator-style GUI? :smile: